geonode/geoserver/helpers.py

```diff
@@ -442,14 +442,14 @@ def set_layer_style(saved_layer, title, sld, base_file=None):
         _old_styles = []
         logger.info(f"SET LAYER STYLE: get old style 1")        
         _old_styles.append(gs_catalog.get_style(
-            name=saved_layer.name))
+            name=saved_layer.name, workspace=saved_layer.workspace))
         logger.info(f"SET LAYER STYLE: get old style 2")        
         _old_styles.append(gs_catalog.get_style(
-            name=f"{saved_layer.workspace}_{saved_layer.name}"))
+            name=f"{saved_layer.workspace}_{saved_layer.name}", workspace=saved_layer.workspace))
```

Comment:
It seems that styles prepended with "geonode_" are both global and in the geonode workspace
![image](https://github.com/geosolutions-it/geonode/assets/717359/cd8f7b50-911a-4066-8afa-b7841a91fd94)